templates/a_muon.cpp.in

```diff
@@ -33,15 +33,236 @@
+ *              is a Fermion and the exchange particle a scalar particle.
+ *
+ * This template evaluates the contribution to the magnetic
+ * dipole moment of a BarZee diagram with fields given by
+ * \a Args.
+ */
+template<class PhotonEmitter, class ExchangeParticle>
+struct AMuonBarZeeFermionLoopZ {
+   static double value(const typename field_indices<Muon>::type& indices,
+                       const context_base& context,
+                       const softsusy::QedQcd& qedqcd);
+};
+
+/**
+ * @class AMuonBarZeeScalarLoop
+ * @brief A template that calculate contributions to the
+ *        anomalous dipole moment of a given particle in
+ *        a BarZee diagram with a scalar loop and scalar exchange
+ *        particle. (for diagram type see arXiv:1502.04199 Figure 2, (2))
+ * @tparam Args Specifies in order the field of which to
+ *              calculate the magnetic dipole moment,
+ *              the photon emitter and the exchange particle
+ *              in a BarZee diagram where the photon emitter
+ *              is a scalar particle and the exchange particle a scalar particle.
+ *
+ * This template evaluates the contribution to the magnetic
+ * dipole moment of a BarZee diagram with fields given by
+ * \a Args.
+ */
+template<class PhotonEmitter, class ExchangeParticle>
+struct AMuonBarZeeScalarLoop {
+   static double value(const typename field_indices<Muon>::type& indices,
+                       const context_base& context,
+                       const softsusy::QedQcd& qedqcd);
+};
+
+/**
+ * @class AMuonBarZeeVectorLoop
+ * @brief A template that calculate contributions to the
+ *        anomalous dipole moment of a given particle in
+ *        a BarZee diagram with a scalar loop and scalar exchange
+ *        particle. (for diagram type see arXiv:1502.04199 Figure 2, (3))
+ * @tparam Args Specifies in order the field of which to
+ *              calculate the magnetic dipole moment,
+ *              the photon emitter and the exchange particle
+ *              in a BarZee diagram where the photon emitter
+ *              is a scalar particle and the exchange particle a scalar particle.
+ *
+ * This template evaluates the contribution to the magnetic
+ * dipole moment of a BarZee diagram with fields given by
+ * \a Args.
+ */
+template<class PhotonEmitter, class ExchangeParticle>
+struct AMuonBarZeeVectorLoop {
+   static double value(const typename field_indices<Muon>::type& indices,
+                       const context_base& context,
+                       const softsusy::QedQcd& qedqcd);
+};
+
+/**
+    Loop function needed for the calculation of the BarZeeDiagram with
+    fermion loop. (arXiv:1502.04199 Equation 26)
+
+    @param respective squared mass ratio.
+*/
+
+double BarZeeLoopFPS(double m)
+{
+
+    if (m == 0)
+        return 0;
+
+    if (m == 0.25)
+        return log(2.0);
+
+    double r1, theta1, r2, theta2;
+    if (m <= 0.25)
+    {
+        double y = std::sqrt(1.0-4.0*m);
```

Comment:
please make `y` a `const` variable (also in the other branches)

Reply:
Done in 275237f8988c508bd2ca7f5aa7126249d4da8b7a.